jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/response/JaxRsResultSerializerFactory.java

```diff
@@ -0,0 +1,100 @@
+package se.fortnox.reactivewizard.jaxrs.response;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import reactor.core.publisher.Flux;
+import se.fortnox.reactivewizard.jaxrs.Stream;
+import se.fortnox.reactivewizard.json.JsonSerializerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+import static reactor.core.publisher.Flux.just;
+
+public class JaxRsResultSerializerFactory {
+    private static final Class<byte[]>   BYTEARRAY_TYPE = byte[].class;
+    private static final Charset         charset        = StandardCharsets.UTF_8;
+
+    private final JsonSerializerFactory jsonSerializerFactory;
+
+    private static final byte[] JSON_ARRAY_START = "[".getBytes();
+    private static final byte[] JSON_ARRAY_END = "]".getBytes();
+
+    private static final byte[] COMMA = ",".getBytes();
+
+    @Inject
+    public JaxRsResultSerializerFactory(JsonSerializerFactory jsonSerializerFactory) {
+        this.jsonSerializerFactory = jsonSerializerFactory;
+    }
+
+    /**
+     * Creates a non streaming result serializer
+     * @param type Content-Type
+     * @param dataCls Return type of the resource method
+     * @param returnTypeIsFlux True if it is a flux, false otherwise
+     * @param <T> Type of elements emitted by the publisher
+     * @return a function that serializes the elements emitted by a publisher to a Flux of byte arrays.
+     */
+    public <T> Function<Flux<T>, Flux<byte[]>> createSerializer(String type, Class<T> dataCls, boolean returnTypeIsFlux) {
+        var byteSerializer = jsonSerializerFactory.createByteSerializer(dataCls);
+        var jsonSerializer = jsonSerializerFactory.createByteSerializer(new TypeReference<>() { });
```

Comment:
`objectMapper.getTypeFactory().constructCollectionType(List.class, dataCls)`

Reply:
Perhaps move these into the correct if-block below to avoid creating serializers that are not needed?